src/main/java/org/springframework/data/elasticsearch/repository/query/ElasticsearchParametersParameterAccessor.java

```diff
@@ -36,13 +38,25 @@ class ElasticsearchParametersParameterAccessor extends ParametersParameterAccess
 	 * @param values must not be {@literal null}.
 	 */
 	ElasticsearchParametersParameterAccessor(ElasticsearchQueryMethod method, Object... values) {
-
 		super(method.getParameters(), values);
 		this.values = Arrays.asList(values);
 	}
 
+	@Override
+	public ElasticsearchParameters getParameters() {
+		return (ElasticsearchParameters) super.getParameters();
+	}
+
 	@Override
 	public Object[] getValues() {
 		return values.toArray();
 	}
+
+	@Override
+	public IndexCoordinates getIndexCoordinatesOrDefaults(@NonNull IndexCoordinates defaults) {
```

Comment:
```suggestion
	public IndexCoordinates getIndexCoordinates(@NonNull IndexCoordinates defaults) {
```

I would not put that in the method name in addition to the parameter name